pkg/operator/credentialsrequest/credentialsrequest_controller_gcp_test.go

```diff
@@ -344,6 +349,61 @@ func TestCredentialsRequestGCPReconcile(t *testing.T) {
 				assert.True(t, cr.Status.Provisioned)
 			},
 		},
+		{
+			name: "cred exists but service account is deleted",
+			existing: []runtime.Object{
+				testOperatorConfig(""),
+				createTestNamespace(testNamespace),
+				createTestNamespace(testSecretNamespace),
+				testProvisionedGCPCredentialsRequest(t),
+				testGCPCredsSecret("kube-system", constants.GCPCloudCredSecretName, testRootGCPAuth),
+				testGCPCredsSecret("openshift-cloud-credential-operator", "cloud-credential-operator-gcp-ro-creds", testReadOnlyGCPAuth),
+				testGCPCredsSecret(testSecretNamespace, testSecretName, testServiceAccountKeyPrivateData),
+				testClusterVersion(),
+				testInfrastructure(testInfraName),
+			},
+			mockRootGCPClient: func(mockCtrl *gomock.Controller) *mockgcp.MockClient {
+				mockGCPClient := mockgcp.NewMockClient(mockCtrl)
+				mockGetProjectName(mockGCPClient)
+
+				// needs update
+				mockGetRole(mockGCPClient)
+				mockListServicesEnabled(mockGCPClient)
+				mockGetDeletedServiceAccount(mockGCPClient)
+
+				// create service account key
+				mockGetRole(mockGCPClient)
+				mockListServicesEnabled(mockGCPClient)
+				mockGetDeletedServiceAccount(mockGCPClient)
+				mockGetProjectIamPolicy(mockGCPClient, testValidPolicyBindings)
+				mockListServiceAccountKeys(mockGCPClient, testServiceAccountKeyName)
+				mockCreateServiceAccount(mockGCPClient)
+				mockSetProjectIamPolicy(mockGCPClient)
+				mockDeleteServiceAccountKey(mockGCPClient, testServiceAccountKeyName)
+				mockCreateServiceAccountKey(mockGCPClient, "NEW AUTH KEY DATA")
+
+				return mockGCPClient
+			},
+			mockReadGCPClient: func(mockCtrl *gomock.Controller) *mockgcp.MockClient {
+				mockGCPClient := mockgcp.NewMockClient(mockCtrl)
+
+				// needs update
+				mockGetProjectName(mockGCPClient)
+				mockGetServiceAccountFailed(mockGCPClient)
+
+				return mockGCPClient
+			},
+			validate: func(c client.Client, t *testing.T) {
+				targetSecret := getCredRequestTargetSecret(c)
+				if assert.NotNil(t, targetSecret) {
```

Comment:
can you just require.NotNil() and avoid putting the checks inside a conditional?

Reply:
This correction needs to be made to other tests too. I will open a separate PR to fix that.